src/Riok.Mapperly/Emit/SyntaxFactoryHelper.cs

```diff
@@ -77,6 +77,8 @@ public static ExpressionSyntax NullSubstitute(ITypeSymbol t, ExpressionSyntax ar
             NullFallbackValue.Default => DefaultLiteral(),
             NullFallbackValue.EmptyString => StringLiteral(string.Empty),
             NullFallbackValue.CreateInstance => CreateInstance(t),
+            _ when argument is ElementAccessExpressionSyntax memberAccess
+                => ThrowNullReferenceException(InterpolatedString($"Sequence {NameOf(memberAccess.Expression)}, contained a null value.")),
```

Comment:
I think including the index in the exception message could help a lot when troubleshooting such issues (eg. `Sequence {nameof(source)} contained a null value at index {index}`.